### PR TITLE
Move some blivet Requires to anaconda.

### PR DESCRIPTION
**Pull request by mulkieran**
When blivet becomes environment aware, it will not need to require these
packages, since it should not fail in the event that these
packages are not installed.

Presumably anaconda still wants the capabilities these packages provide,
since it is used to having them through blivet, so add these Requires to
anaconda.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

File: anaconda.spec.in [hunk 1]
```diff
@@ -145,6 +146,10 @@ Requires: newt-python
 # required because of the rescue mode and VNC question
 Requires: anaconda-tui = %{version}-%{release}
 
+# Storage packages that blivet considers optional
+Requires: libblockdev-plugins-all >= %{libblockdevver}
```
Comment (vpodzime) [inline anaconda.spec.in]: The idea behind this is that the ``libblockdev`` package will still be required by Blivet, right? I think we need some comment about it in the spec.

**Comment by vpodzime** [comment]
Generally, I'm not sure how to approach this from the right direction -- e.g. when I change something in blivet that requires new libblockdev with this I'll have to go to anaconda's spec and bump the required version of ``libblockdev`` there. That seems like a perfect thing to forget.

I know many guys don't like subpackages, but what about adding ``blivet-all` as a subpackage of Blivet pulling in blivet and all its external dependencies? That way we would keep track of Blivet's external dependencies in Blivet's spec where it really belongs and we will have an easy way of installing them all into the installation images.

**Comment by bcl** [comment]
I think the right way to do it is remove the version number and have anaconda depend on just the libblockdev-plugins-all package since it doesn't really care what the specific version is. blivet should enforce the specific version that it needs by a dependency on the main package, which I think will cause the correct plugin versions to be pulled in. No need for subpackages.

**Comment by vpodzime** [comment]
> I think the right way to do it is remove the version number and have anaconda depend on just the libblockdev-plugins-all package since it doesn't really care what the specific version is. blivet should enforce the specific version that it needs by a dependency on the main package, which I think will cause the correct plugin versions to be pulled in. No need for subpackages.

How should that happen? The ``libblockdev`` (main) package doesn't require the plugins and thus cannot require any specific version(s).

**Comment by mulkieran** [comment]
Will the versions of the plugins and the version of libblockdev proper always be so tightly coupled as they are now? If not, then this problem will soon look a bit different and updating the plugin version(s) separately from the library will seem exactly right.

**Comment by vpodzime** [comment]
> Will the versions of the plugins and the version of libblockdev proper always be so tightly coupled as they are now? If not, then this problem will soon look a bit different and updating the plugin version(s) separately from the library will seem exactly right.

One of the key ideas behind the complicated design of a library with a lot of plugins is that it allows for better granularity of updates so the above is an exactly valid concern. Once it stabilizes a bit the ``libblockdev`` package (the library) should be stable and unchanged unless new plugins are added.

**Comment by bcl** [comment]
> How should that happen? The libblockdev (main) package doesn't require the plugins and thus cannot require any specific version(s).

I think (not having tried this) that by blivet requiring a newer libblockdev the plugins will be updated to the latest available version.

**Comment by mulkieran** [comment]
Update to previous version. Added a few lines to avoid displaying unsupported device types and so forth. These will only change behaviour if a dependency is missing, which is generally unlikely for anaconda.

I'm sticking w/ the current way of handling libblockdev plugin dependencies, I think it is the best and able to adapt best to when library becomes stable.

**Comment by mulkieran** [comment]
"Only show supported autopart..." has been revised a bit, otherwise no changes.

**Comment by vpodzime** [comment]
All the code changes look good to me now, thanks! I just still don't like the dependencies moved to *anaconda* this way -- *blivet-gui*, *openlmi* and in future all other packages using *blivet* would need to do the same. And it just doesn't make sense that if blivet's functionality didn't work because of a bug in libblockdev, the projects using **blivet** would have to change their requires. These projects are using **blivet**'s API and thus should only track required versions of *blivet*.

What about adding a ``blivet-core`` subpackage to *blivet* and having ``blivet`` as ``blivet-core + libblockdev-plugins-all``? That wouldn't require any changes outside of blivet for all the projects that use it and don't want to use only some pieces of its functionality.

**Comment by mulkieran** [comment]
I understand the wish to handle blivet's dependencies in blivet...hence the various subpackage proposals.

The problem is that while they allow blivet to keep track of versioning they give less benefit to clients of blivet. These clients are stuck in an all or nothing situation, either they require blivet and all libblockdev plugins and dosfstools or they require blivet alone and have to do all the management of other requires themselves.

Why not have blivet individually Suggest plugins and other optional dependencies, along w/ versions, since ultimately plugins may diverge? Make a single package or meta-package which requires everything that blivet only suggests _and_ that anaconda will want to require, e.g., all libblockdev plugins but not gfs2-utils. Call it blivet-anaconda. Maybe blivet-gui will want a different set of dependencies...call that blivet-blivet-gui. Maybe blivet-gui will want a different set of dependencies depending on what environment it thinks it will run in. Call its meta-packages blivet-blivet-gui-1, blivet-blivet-gui-2...

Encourage any other clients of blivet that want a different subset of the tools that we suggest to submit meta-packages with the same general naming scheme. I don't see why we should have any difficulty in building these meta-packages.

I don't think this is a radically different suggestion from the two subpackage suggestions above. The most significant difference is that it emphasizes that anaconda is just one of several clients of blivet, and, because it really needs to be versatile, one that wants to turn a lot of blivet suggests into requires.

It might make sense to add a few more suggests than there were requires in the blivet spec file originally, e.g., for gfs2-tools.

**Comment by mulkieran** [comment]
@vpodzime, @dwlehman: How about I just push the code patches, and the question of dependencies can be worked out separately.

**Comment by vpodzime** [comment]
> How about I just push the code patches, and the question of dependencies can be worked out separately.

Works for me.

**Comment by dwlehman** [comment]
Sounds good to me.

**Pull request merged**